packages/spartan/extra/dcos-spartan.service

```diff
@@ -19,6 +19,6 @@ ExecStartPre=/usr/bin/env ip link set spartan up
 ExecStartPre=-/usr/bin/env ip addr add 198.51.100.1/32 dev spartan
 ExecStartPre=-/usr/bin/env ip addr add 198.51.100.2/32 dev spartan
 ExecStartPre=-/usr/bin/env ip addr add 198.51.100.3/32 dev spartan
-ExecStart=/opt/mesosphere/active/spartan/spartan/bin/spartan-env foreground
+ExecStart=/usr/bin/env flock -x /opt/mesosphere/active/spartan/spartan/bin/spartan-env /opt/mesosphere/active/spartan/spartan/bin/spartan-env foreground
```

Comment:
Nothing is going to write to spartan-env. The purpose of the flock is to signal that Spartan is running. 

The difference of exec start Pre, would be instead of:
```
ExecStart=/opt/mesosphere/bin/dcos-spartan-watchdog.sh
```

it'd be:

```
ExecStartPre=/bin/bash -c "if flock -n /opt/mesosphere/active/spartan/spartan/bin/spartan-env; then exit 1; fi"
ExecStart=/opt/mesosphere/active/toybox/bin/toybox timeout -k 1m 1m /bin/bash -c "/opt/mesosphere/active/toybox/bin/toybox host ready.spartan 198.51.100.1 || /opt/mesosphere/active/toybox/bin/toybox pkill -l 9 -f spartan"		
```